src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Calculator/CalculateHelper.cs

```diff
@@ -60,6 +64,37 @@ public static string FixHumanMultiplicationExpressions(string input)
             return output;
         }
 
+        private static string CheckScientificNotation(string input)
+        {
+            /**
+             * NOTE: Anywhere you see a "{0}", that is meant to be replaced by the user's decimal separator.
+             *
+             * Regex explanation:
+             * (-?(\d+({0}\d*)?)|-?({0}\d+)): Used to capture one of two types:
+             * -?(\d+({0}\d*)?): Captures a decimal number starting with a number (e.g. "-1.23")
+             * -?({0}\d+): Captures a decimal number without leading number (e.g. ".23")
+             * E: Captures capital 'E'
+             * (-?\d+{0}?\d*): Captures decimal number starting with a number (e.g. "1.23")
+             */
+            var p = @"(-?(\d+({0}\d*)?)|-?({0}\d+))E(-?\d+{0}?\d*)";
+
+            // If the decimal separator is a ".", we also have to insert a backslash.
+            if (CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator == ".")
+            {
+                p = string.Format(CultureInfo.CurrentCulture, p, @"\.");
+            }
+            else
+            {
+                p = string.Format(
+                    CultureInfo.CurrentCulture,
+                    p,
+                    CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator);
+            }
```

Comment:
Technically, this can also be done with a ternary. But both would work.

Reply:
I thought a ternary would take too much horizontal space because the `CultureInfo.CurrentCulture...` is a lot of characters so I left it at the expanded if/else instead